pandas/tests/indexing/test_loc.py

```diff
@@ -1114,24 +1145,110 @@ def test_loc_setitem_df_datetime64tz_column_without_index():
     )
     df2 = pd.DataFrame(index=df.index)
     df2.loc[:, "data"] = df["data"]
-    tm.assert_series_equal(df.data, df2.data)
+    tm.assert_frame_equal(df, df2)
 
 
-def test_loc_setitem_series_datetime64tz_with_index():
+def test_loc_setitem_series_datetime64tz_full_with_index():
     s1 = pd.Series(
         pd.date_range("2020-01-01", "2020-01-06", 6, tz=timezone.utc), name="data"
     )
-    s2 = pd.Series(index=s1.index, dtype=np.object, name="data")
+    s2 = pd.Series(index=s1.index, dtype="object", name="data")
     s2.loc[s1.index] = s1
-    tm.assert_numpy_array_equal(np.array(s1), np.array(s2))
-    assert s2.dtype == np.object
+    tm.assert_series_equal(s1, s2)
+
+
+def test_loc_setitem_series_datetime64tz_slice():
+    s1 = pd.Series(
+        pd.date_range("2020-01-01", "2020-01-06", 6, tz=timezone.utc), name="data"
+    )
+    s2 = pd.Series(index=s1.index, dtype="object", name="data")
+    expected = pd.Series(
+        [
+            *pd.date_range("2020-01-01", "2020-01-03", 3, tz=timezone.utc),
```

Comment:
why the "*"

Reply:
i guess do

```
dti = pd.date_range("2020-01-01", "2020-01-03", periods=6, tz=timezone.utc)
expected = pd.Series(dti)
expected[-3:] = pd.NaT
```

Reply:
@jbrockmendel okay, if that is preferred.

Reply:
@jbrockmendel please note the `object` `dtype` for the result of a slice assignement